micro-rdk/src/common/conn/server.rs

```diff
@@ -411,19 +426,14 @@ where
             }
         }
     }
-    async fn serve_http2<U>(
-        &self,
-        mut c: U,
+    async fn serve_http2(
+        connection: T,
+        exec: Executor,
         robot: Arc<Mutex<LocalRobot>>,
-    ) -> Result<(), ServerError>
-    where
-        U: Http2Connector<Stream = T>,
-    {
+    ) -> Result<(), ServerError> {
         let srv = GrpcServer::new(robot.clone(), GrpcBody::new());
-        let connection = c.accept().await.map_err(|e| ServerError::Other(e.into()))?;
-
         Box::new(
-            http2::Builder::new(self.exec.clone())
+            http2::Builder::new(exec)
                 .initial_connection_window_size(2048)
                 .initial_stream_window_size(2048)
                 .max_send_buf_size(4096)
```

Comment:
Would it be better to move this functionality to [here](https://github.com/acmorrow/micro-rdk/blob/45c9de1f4efc39cac7691a14d1887788ddc6c082/micro-rdk/src/common/conn/server.rs#L505)?

Run can take an enum with the appropriate arguments. something like `c.run(Serve::Http{exec, robot})`

Reply:
Maybe, but I often try to avoid moving code while also changing it. It makes it much harder to review. I think I'd like to keep it here for now until I've finished iterating on `ViamServer::serve`. I'll add a TODO.

Reply:
sounds good. maybe tag it with `good-first-issue`. we'll need more of those kinds of tasks soon